api/V1/Schedule.js

```diff
@@ -0,0 +1,151 @@
+const models       = require('../../models');
+const responseUtil = require('./responseUtil');
+const middleware   = require('../middleware');
+const { body }     = require('express-validator/check');
+
+module.exports = (app, baseUrl) => {
+  var apiUrl = baseUrl + '/schedules';
+
+  /**
+   * @api {post} /api/v1/schedules Adds a new schedule
+   * @apiName PostSchedule
+   * @apiGroup Schedules
+   * @apiDescription This call is used to upload a new audio bait schedule which controls when
+   * sound bait files are played.
+   *
+   * @apiUse V1UserAuthorizationHeader
+   *
+   * @apiBody {JSON} data Metadata about the schedule in JSON format.
+   *
+   * @apiParam {Int[]} devices List of device Ids that schedule should apply to
+   * @apiParam {JSON} schedule Schedule
+   *
+   * @apiUse V1ResponseSuccess
+   * @apiuse V1ResponseError
+   */
+  app.post(
+    apiUrl,
+    [
+      middleware.authenticateUser,
+      body('schedule', '"schedule" is missing').exists(),
+      body('devices', '"devices" is missing.  This is a list of device ids that the schedule should be applied to.').exists(),
+    ],
+    middleware.requestWrapper(async function(request, response) {
+      var deviceIds = request.body.devices;
+      try {
+        await request.user.checkUserControlsDevices(deviceIds);
+      }
+      catch (error) {
+        if (error.name == 'UnauthorizedDeviceException') {
+          return responseUtil.send(response, {
+            statusCode: 400,
+            success: false,
+            messages: [error.message]
+          });
+        } else {
+          throw error;
+        }
+      }
+
+      var instance = models.Schedule.build(request.body, ["schedule"]);
+      instance.set('UserId', request.user.id);
+      await instance.save();
+
+      await models.Device.update(
+        {ScheduleId: instance.id},
+        {where: {id: deviceIds}}
+      );
```

Comment:
It would be better if the insertion of the schedule and the device schedule updates were done in a single database transaction.